operator/mapper/dbaas.go

```diff
@@ -31,19 +31,20 @@ func set(s []string) map[string]struct{} {
 	return m
 }
 
-func CompareSettings(a, b runtime.RawExtension) bool {
-	sa, err := ToMap(a)
+// CompareSettings compares 2 instances of service settings
+// Some DBaaS add default settings to their services, which lead to unequal settings between
+// managed and provider's resource
+// Ignore added provider configuration
+func CompareSettings(managed, provider runtime.RawExtension) bool {
+	sa, err := ToMap(managed)
 	if err != nil {
 		// we have to assume they're not the same
 		return false
 	}
-	sb, err := ToMap(b)
+	sb, err := ToMap(provider)
 	if err != nil {
 		return false
 	}
-	if len(sa) != len(sb) {
-		return false
-	}
```

Comment:
This is technically not correct. We're now unable to remove settings.

Let's say we set a setting `bar` and later we remove that setting. This function now won't detect the change and will not revert the setting to the default

Reply:
The best I could come up with for kafka rest settings was this... https://github.com/vshn/provider-exoscale/blob/master/operator/kafkacontroller/observe.go#L213-L247

Maybe you have a better idea :sweat_smile:

Reply:
I haven't noticed that part, I was wondering why you are using Rest settings :)) As you noticed my solution was very easy and very wrong. Lets see, maybe I can come up with something tomorrow.

Reply:
Now that I'm checking it better I don't thing it's a very good approach hardcoding the defaults that arrive from exoscale... with the next version this might change

Reply:
Yeah it's not a great approach, but I don't have a better one

Reply:
Your approach is the only one that's working therefore it is the best period.